frontend/src/Editor/QueryManager/QueryManager.jsx

```diff
@@ -801,6 +807,7 @@ class QueryManagerComponent extends React.Component {
                     components={this.props.allComponents}
                     apps={this.props.apps}
                     popoverPlacement="top"
+                    pages={this.state.pages}
```

Comment:
@arpitnath Can we access it through `props` directly instead of setting `state`?